src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -735,40 +735,42 @@ internal static void UpdateLayerBorder(this CoreAnimation.CALayer layer, IButton
 			return null;
 		}
 
-		internal static UIView? FindNextView(this UIView view, UIView superView, Func<UIView, bool> isValidType)
+		internal static UIView? FindNextView(this UIView view, UIView containerView, Func<UIView, bool> isValidType)
 		{
-			var passedOriginal = false;
+			UIView? nextView = null;
 
-			var nextView = superView.FindNextView(view, ref passedOriginal, isValidType);
+			while (view != containerView && view is not null)
+			{
+				var siblings = view.Superview.Subviews;
+				nextView = view.FindNextView(siblings.IndexOf(view) + 1, isValidType);
+
+				if (nextView is not null)
+					break;
+
+				view = view.Superview;
+			}
 
 			// if we did not find the next view, try to find the first one
-			nextView ??= superView.FindNextView(null, ref passedOriginal, isValidType);
+			nextView ??= containerView.Subviews[0].FindNextView(0, isValidType);
```

Comment:
Is it possible for there to be 0 `Subviews` here? Would it throw an exception?

Reply:
According the docs it is not a nullable property, but I can add a check just in case since I was a little weary of that as well!

Reply:
Actually yeah even if subviews is not null, the zero index could be so better to handle that! Thanks!

Reply:
Okay updated accordingly @jonathanpeppers